src/goose.rs

```diff
@@ -908,9 +921,13 @@ impl GooseUser {
     ///       let _goose = user.post("/path/to/foo/", "BODY BEING POSTED").await;
     ///     }
     /// ```
-    pub async fn post(&self, path: &str, body: &str) -> GooseResponse {
+    pub async fn post(
+        &self,
+        path: &str,
+        body: &str,
+    ) -> Result<GooseResponse, mpsc::error::SendError<bool>> {
         let request_builder = self.goose_post(path).await.body(body.to_string());
-        self.goose_send(request_builder, None).await
+        Ok(self.goose_send(request_builder, None).await?)
```

Comment:
Interesting pattern, but I guess it's completely identical to:

```
let result = self.goose_send(request_builder, None).await?;
Ok(result)
```

which is:

```
let result = match self.goose_send(request_builder, None).await {
  Err(e) => return Err(e),
  Ok(g) => g,
}

Ok(result)
```

So yeah - makes sense now :) - was just new for me.

Reply:
Is it overly compact? I'm not a fan of the shortest way to write something if it makes it harder to read, and would not have any problem splitting it out so we return Ok(result) instead.

Reply:
For me:

```
let result = self.goose_send(request_builder, None).await?;
Ok(result)
```

is easier to read, but that's a question for a rust person really ...

Reply:
I will leave as-is for now, we can address in a followup if we decide to split it.